oidc-wp/openid-connect-generic
# fix: #458 Log Cleanup When Log Limit Changed

### All Submissions:






### Changes proposed in this Pull Request:



* Updates the minimum PHP version to 7.4.
* Updates the development PHP version to 8.0.
* Updates the development dependencies to the latest versions.
* Updates additional linting & static analysis dependencies.
* Fixes newly identified linting & static analysis errors.
* Fixes #458.
* Updates the log cleanup handling to properly retain the configured number of log entries.
* Updates the log display output to reflect the log retention policy.

### How to test the changes in this Pull Request:

1. Login to the WordPress Dashboard.
2. Turn on the plugin logging.
3. Configure the desired log entries retention.
4. Access the site login page to observe log entries being created.
5. Observe that after hitting the desired retention length old record roll off with new one being added.
6. Change the log retention number.
7. Observe that the logs listed reflect the new log limit if it is smaller.

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
* Updates the log cleanup handling to properly retain the configured number of log entries.
* Updates the log display output to reflect the log retention policy.
